enterprise/cli/licenses.go

```diff
@@ -0,0 +1,111 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+
+	agpl "github.com/coder/coder/cli"
+	"github.com/coder/coder/cli/cliui"
+	"github.com/coder/coder/codersdk"
+)
+
+var jwtRegexp = regexp.MustCompile(`^[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+$`)
+
+func licenses() *cobra.Command {
+	cmd := &cobra.Command{
+		Short:   "Add, remove, and list licenses",
+		Use:     "licenses",
+		Aliases: []string{"license"},
+	}
+	cmd.AddCommand(
+		licenseAdd(),
+	)
+	return cmd
+}
+
+func licenseAdd() *cobra.Command {
+	var (
+		filename string
+		license  string
+		debug    bool
+	)
+	cmd := &cobra.Command{
+		Use:  "add",
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := agpl.CreateClient(cmd)
+			if err != nil {
+				return err
+			}
+			if filename != "" && license != "" {
+				return xerrors.New("only one of (--filename, --license) may be specified")
+			}
+
+			if filename == "" && license == "" {
```

Comment:
Suggestion: Consider simplifying with an `if/else` or `switch` since we already do the check above.

```suggestion
			if license == "" {
```

Reply:
I think you still need both?

```golang
filename = "file.json"
license = ""
```

`filename != "" && license != ""` is `false`
`filename == "" && license == ""` is `false`
`license == ""` is `true`

----

I like the idea of a switch statement too.

Reply:
Oof, thanks for being more observant than me @Emyrk !